consensus/beacon/consensus.go

```diff
@@ -403,8 +403,18 @@ func (beacon *Beacon) FinalizeAndAssemble(chain consensus.ChainHeaderReader, hea
 	// Assign the final state root to header.
 	header.Root = state.IntermediateRoot(true)
 
+	if chain.Config().IsOptimismIsthmus(header.Time) {
+		if body.Withdrawals == nil || len(body.Withdrawals) > 0 { // We verify nil/empty withdrawals in the CL pre-Isthmus
+			return nil, fmt.Errorf("expected non-nil empty withdrawals operation list in Isthmus, but got: %v", body.Withdrawals)
+		}
+		// State-root has just been computed, we can get an accurate storage-root now.
+		h := state.GetStorageRoot(params.OptimismL2ToL1MessagePasser)
+		header.WithdrawalsHash = &h
+		state.AccessEvents().AddAccount(params.OptimismL2ToL1MessagePasser, false) // include in execution witness
```

Comment:
Why is this included in the execution witness? Shouldn't it only be in the witness if it changed? Its not being touched as part of EVM execution

Reply:
I added it in there a while ago, because the storage-root isn't otherwise available for block-header verification. I figured it's better to add it explicitly now, than have to debug why it's failing to do verkle/stateless/other just because of a missing access.